app/controllers/variables_controller.rb

```diff
@@ -118,4 +119,21 @@ def set_schema
   def set_form
     @json_form = UmmJsonForm.new('umm-var-form.json', @schema, @variable, 'field_prefix' => 'variable_draft/draft')
   end
+
+  def ensure_correct_variable_provider
+    return if @provider_id == current_user.provider_id
+
+    @variable_action = if request.original_url.include?('edit')
+                         'edit'
+                       elsif request.original_url.include?('delete')
+                         'delete'
+                       end
+
+    @user_permissions = 'none'
+    if current_user.available_providers && current_user.available_providers.include?(@provider_id)
```

Comment:
Where you do these provider checks, I made some helper methods in `app/helpers/users_helper.rb` that you could also use.